src/internal.c

```diff
@@ -24591,6 +24800,15 @@ int SendClientKeyExchange(WOLFSSL* ssl)
                 }
                 ssl->options.clientState = CLIENT_KEYEXCHANGE_COMPLETE;
             }
+        #if defined(OPENSSL_EXTRA) && defined(HAVE_SECRET_CALLBACK)
+            if (ssl->sessionSecretCb != NULL) {
+                int secretSz = SECRET_LEN;
+                ret = ssl->sessionSecretCb(ssl, ssl->arrays->masterSecret,
```

Comment:
Why is session secret cb used for the key log cb?  How do you know the key log cb is even set up?  What if the user wants both?

Why is the key log cb only called here on client key exchange send?  There are several places where the client and server generate key material.  Where are those handled?

Reply:
The session secret callback doesn't seem to be OpenSSL compatible, but I thought the keylog cb and session secret cb were provided for almost the same purpose. So I thought we should use the session secret Cb function as much as possible. It was confirmed that it works by connecting the key log Cb to the existing session secret Cb. There are two problems with this strategy:
1. As you pointed out, the session secret Cb and key log Cb that are set later have priority. However, customers who want to set both in the same program are likely to be convinced.
2. The call points for the existing session secret Cb are at DoServerHello () internal.c and DoTls13ServerHello () in tls13.c. Unfortunately, no masterSecret was generated at this call timing. So I needed to create a new call point, so I chose this SendClientKeyExchange (). And other materials are handled properly with existing session secret Cb call points.

Is the above the answer to your question?